### PR TITLE
[lgwebos] Add dynamic options providers for application and channel

### LINKED ISSUE #7123: [lgwebos] Dynamic option providers
Dynamic option providers could be added for the following channels:

- Applicatiuon
- Channel

This will make easy a selection amongst available applications and channels in any UI.

I will implement this enhancement.

### PR DESCRIPTION
**Pull request by lolodomo**
Fixes #7123

Signed-off-by: Laurent Garnier <lg.hc@free.fr>

### CODE REVIEW

**Comment by lolodomo** [comment]
@sprehn  : please note that with this PR the channel "channelName" is now useless and if you agree, I can remove it and just keep the channel "channel".

**Comment by lolodomo** [comment]
Here is the result in Basic UI (when using Defaut in sitemap):
image
image

**Comment by sprehn** [comment]
Hi,

this is Option Provider is a new concept to me and I like it. 
do we need to update sth in Action?  e.g. Would it be possible to loose Application in the action package as well?

**Comment by lolodomo** [comment]
> 
> do we need to update sth in Action? e.g. Would it be possible to loose Application in the action package as well?

to loose ? I am sorry, I don't understand your question.

**Comment by sprehn** [comment]
In the package for action there is a class called Application, which is only carries id and name and is used as a return type in List<Application> getApplications.
It is pretty much the same as this Option.

But no big deal, if not.

File: bundles/org.openhab.binding.lgwebos/README.md [hunk 1]
```diff
@@ -106,12 +106,12 @@ Switch LG_TV0_Power "TV Power" <television>  { autoupdate="false", channel="lgwe
 Switch LG_TV0_Mute  "TV Mute"                { channel="lgwebos:WebOSTV:3aab9eea-953b-4272-bdbd-f0cd0ecf4a46:mute"}
 Dimmer LG_TV0_Volume "Volume [%d]"           { channel="lgwebos:WebOSTV:3aab9eea-953b-4272-bdbd-f0cd0ecf4a46:volume" }
 Number LG_TV0_VolDummy "VolumeUpDown"
-String LG_TV0_Channel "Channel [%s]"         { channel="lgwebos:WebOSTV:3aab9eea-953b-4272-bdbd-f0cd0ecf4a46:channel" }
+Default LG_TV0_Channel "Channel [%s]"        { channel="lgwebos:WebOSTV:3aab9eea-953b-4272-bdbd-f0cd0ecf4a46:channel" }
```
Comment (cpmeister) [inline bundles/org.openhab.binding.lgwebos/README.md]: What is a "Default" item type?

Comment (lolodomo) [reply bundles/org.openhab.binding.lgwebos/README.md]: Mistake ! The default must be in the sitemap file, not in the item file.

Comment (lolodomo) [reply bundles/org.openhab.binding.lgwebos/README.md]: Fixed now

**Comment by cpmeister** [comment]
Manually verified sign.off

**Comment by sprehn** [comment]
@lolodomo I am trying to replicate the setup you showed on your screenshot with those new options and changed sitemap accordingly to Default

I only get the id value as text for Application (bottom left) or Kanal No (second line left)

Bildschirmfoto 2020-03-10 um 23 43 39

any idea what I did wrong?

**Comment by lolodomo** [comment]
I just compiled a new version from branch 2.5.x and it still works for me.
Take care to clear the cache of your browser.
Here is how are defined my items:
```
String TvNumChaine "Chaîne TV [%s]" <none> (GTvSalon) { channel="lgwebos:WebOSTV:tv:channel" }
String TvApplication "Application [%s]" <none> (GTvSalon) { channel="lgwebos:WebOSTV:tv:appLauncher" }
```
and my sitemap:
```
			Default item=TvNumChaine
			Default item=TvApplication
```

**Comment by lolodomo** [comment]
@sprehn : do you agree with me that the channel `channelName` is now useless ?
I propose to suppress it and to clarify the README that for the channel `channel`, the command value remains the channel number.

**Comment by sprehn** [comment]
> @sprehn : do you agree with me that the channel channelName is now useless ?

yes, fully agree

**Pull request merged**